File: src/main/java/io/wollinger/snipsniper/utils/InputContainer.java
package io.wollinger.snipsniper.utils;

import java.awt.*;
import java.util.ArrayList;

public class InputContainer {
    private final boolean[] keys = new boolean[9182];
    private int mouseX;
    private int mouseY;

    private final ArrayList<Point> mousePath = new ArrayList<>();

    public void setKey(int keyCode, boolean pressed) {
        keys[keyCode] = pressed;
    }

    public void setMousePosition(int mouseX, int mouseY) {
        this.mouseX = mouseX;
        this.mouseY = mouseY;
    }

    public void addMousePathPoint(Point point) {
        mousePath.add(point);
    }

    public boolean isKeyPressed(int keyCode) {
        return keys[keyCode];
    }

    public boolean areKeysPressed(int... keyCodes) {
        for(int n : keyCodes) {
            if(!keys[n])
                return false;
        }
        return true;
    }

    public int getMouseX() {
        return mouseX;
    }

    public int getMouseY() {
        return mouseY;
    }

    public Point getMousePathPoint(int i) {
        if(mousePath.size() > i)
            return mousePath.get(i);
        return null;
    }

    public void removeMousePathPoint(int i) {
        mousePath.remove(i);
    }

    public void clearMousePath() {
        mousePath.clear();
    }

}

File: src/main/kotlin/net/snipsniper/utils/InputContainer.kt
package net.snipsniper.utils

import java.awt.Point
import java.util.*
import kotlin.collections.ArrayList

class InputContainer {
    private val keys = BooleanArray(9182)
    private val mousePath = ArrayList<Point>()
    var mouseX = 0
    var mouseY = 0

    fun setMousePosition(mouseX: Int, mouseY: Int) {
        this.mouseX = mouseX
        this.mouseY = mouseY
    }
    fun getMousePoint(): Point = Point(mouseX, mouseY)

    fun addMousePathPoint(point: Point) = mousePath.add(point)
    fun getMousePathPoint(i: Int): Point? = mousePath.getOrNull(i)
    fun removeMousePathPoint(i: Int) = mousePath.removeAt(i)
    fun clearMousePath() = mousePath.clear()

    fun resetKeys() = Arrays.fill(keys, false)
    fun setKey(keyCode: Int, pressed: Boolean) { keys[keyCode] = pressed }
    fun isKeyPressed(keyCode: Int): Boolean = keys[keyCode]
    fun areKeysPressed(vararg keyCodes: Int): Boolean {
        keyCodes.forEach { if(!keys[it]) return false }
        return true
    }
}